test/e2e/specs/editor/blocks/list.spec.js

```diff
@@ -0,0 +1,872 @@
+		await page.keyboard.type( '* ' );
+		await page.evaluate( () => new Promise( window.requestIdleCallback ) );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'ArrowDown' );
+		await page.keyboard.press( 'Backspace' );
+
+		expect( await editor.getEditedPostContent() ).toBe( '' );
+	} );
+
+	test( 'can be created by typing "/list"', async ( { editor, page } ) => {
+		// Create a list with the slash block shortcut.
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+		await page.keyboard.type( '/list' );
+		await expect(
+			page.locator( 'role=option[name="List"i][selected]' )
+		).toBeVisible();
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'I’m a list' );
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:list -->
+<ul><li>I’m a list</li></ul>
+<!-- /wp:list -->`
+		);
+	} );
+
+	test( 'can be created by converting a paragraph', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+		await page.keyboard.type( 'test' );
+		await editor.transformBlockTo( 'List' );
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:list -->
+<ul><li>test</li></ul>
+<!-- /wp:list -->`
+		);
+	} );
+
+	test( 'can be created by converting multiple paragraphs', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+		await page.keyboard.type( 'one' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'two' );
+		await page.keyboard.down( 'Shift' );
+		await page.click( '[data-type="core/paragraph"]:first-of-type' );
+		await page.keyboard.up( 'Shift' );
+		await editor.transformBlockTo( 'List' );
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:list -->
+<ul><li>one</li><li>two</li></ul>
+<!-- /wp:list -->`
+		);
+	} );
+
+	test( 'can be created by converting a paragraph with line breaks', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+		await page.keyboard.type( 'one' );
+		await pageUtils.pressKeyWithModifier( 'shift', 'Enter' );
+		await page.keyboard.type( 'two' );
+		await editor.transformBlockTo( 'List' );
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:list -->
+<ul><li>one</li><li>two</li></ul>
+<!-- /wp:list -->`
+		);
+	} );
+
+	test( 'should not transform lines in block when transforming multiple blocks', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+		await page.keyboard.type( 'one' );
+		await pageUtils.pressKeyWithModifier( 'shift', 'Enter' );
+		await page.keyboard.type( '...' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'two' );
+		await page.keyboard.down( 'Shift' );
+		await page.click( '[data-type="core/paragraph"]:first-of-type' );
```

Comment:
Let's use the [N-th selector](https://playwright.dev/docs/selectors#n-th-element-selector) instead:

```suggestion
		await page.click( '[data-type="core/paragraph"] >> nth=0' );
```

Same for the other occurrences below.